src/cljs/org/broadinstitute/firecloud_ui/main.cljs

```diff
@@ -321,66 +330,130 @@
                  " for more information."])}))
 
 
+(react/defc ConfigLoader
+  {:render
+   (fn [{:keys [state]}]
+     [:div {:style {:padding "40px 0"}}
+      (if (:error? @state)
+        [:div {:style {:color (:exception-state style/colors)}}
+         "Error loading configuration. Please try again later."]
+        [comps/Spinner {:text "Loading configuration..."}])])
+   :component-did-mount
+   (fn [{:keys [props state]}]
+     ;; Use basic ajax call here to bypass authentication.
+     (utils/ajax {:url "/config.json"
+                  :on-done (fn [{:keys [success? get-parsed-response]}]
+                             (if success?
+                               (do
+                                 (reset! config/config (get-parsed-response false))
+                                 ((:on-success props)))
+                               (swap! state assoc :error true)))}))})
```

Comment:
That is a fantastic idea. To keep this PR clean, I'll do that as a follow-up (GAWB-1334).